integration/test/metric/cpu.go

```diff
@@ -0,0 +1,72 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build linux && integration
+// +build linux,integration
+
+package metric
+
+import (
+	"log"
+
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch/types"
+	"github.com/aws/aws-sdk-go/aws"
+)
+
+type CPUMetricValueFetcher struct {
+	base *baseMetricValueFetcher
+}
+
+func (f *CPUMetricValueFetcher) Fetch(namespace string, metricName string, stat Statistics) ([]float64, error) {
+	dimensions := f.getMetricSpecificDimensions()
+	values, err := f.fetch(namespace, dimensions, metricName, stat)
+	if err != nil {
+		log.Printf("Error while fetching metric value for %v: %v", metricName, err.Error())
+	}
+	return values, err
+}
+
+func (f *CPUMetricValueFetcher) fetch(namespace string, metricSpecificDimensions []types.Dimension, metricName string, stat Statistics) ([]float64, error) {
+	return f.base.fetch(namespace, metricSpecificDimensions, metricName, stat)
+}
```

Comment:
I don't see the point of overriding the base class implementation here.

I think it would be simpler just to have:
```
type CPUMetricValueFetcher struct {
    baseMetricValueFetcher
}
```
Then you do not need to override.
You can just remove:
```
func (f *CPUMetricValueFetcher) fetch(namespace string, metricSpecificDimensions []types.Dimension, metricName string, stat Statistics) ([]float64, error) {
	return f.base.fetch(namespace, metricSpecificDimensions, metricName, stat)
}
```
And any call to `CPUMetricValueFetcher.fetch()` will call the base class's implementation.

more details - https://golangbot.com/inheritance/

Reply:
I think the issue is the pointer reference in the struct, but I'm not entirely sure. I haven't wrapped my head fully around inheritance in Go yet

Reply:
yeah I mean this is technically not inheritance. They consider this composition and while I'm loving that go is forcing me to always use composition rather than inheritance, I don't know how this would behave(even if it works) if I take in two structs that both implement the same method e.g. baseMetricFetcher.fetch() and complexMetricFetcher.fetch() both available

Reply:
ok this built.. I think I wasn't understanding the syntax yesterday or something. Waiting for the test to run

Reply:
> I don't know how this would behave(even if it works) if I take in two structs that both implement the same method
I know what happens now. This was the problem that I had to fix yesterday.

Verified things work with the tests after the change